cmd/byohctl/client/k8s.go

```diff
@@ -0,0 +1,237 @@
+		return nil, utils.LogErrorf("error creating request: %v", err)
+	}
+
+	req.Header.Add("Authorization", "Bearer "+c.bearerToken)
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, utils.LogErrorf("error making request: %v", err)
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, utils.LogErrorf("error reading response: %v", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, utils.LogErrorf("error getting secret (status %d): %s", resp.StatusCode, string(body))
+	}
+
+	var secret types.Secret
+	err = json.Unmarshal(body, &secret)
+	if err != nil {
+		return nil, utils.LogErrorf("error parsing secret: %v", err)
+	}
+
+	utils.LogSuccess("Successfully retrieved secret")
+	return &secret, nil
+}
+
+// SaveKubeConfig saves the kubeconfig from the secret to disk
+func (c *K8sClient) SaveKubeConfig(secretName string) error {
+	// Step 1: Get secret
+	secret, err := c.GetSecret(secretName)
+	if err != nil {
+		return fmt.Errorf("failed to get secret: %v", err)
+	}
+	utils.LogSuccess("Successfully retrieved secret")
+
+	// Step 2: Get kubeconfig from secret
+	kubeconfigString, ok := secret.Data["config"]
+	if !ok {
+		return fmt.Errorf("kubeconfig not found in secret (missing 'config' key)")
+	}
+
+	// Step 3: Decode the base64 encoded kubeconfig
+	decodedKubeconfig, err := base64.StdEncoding.DecodeString(kubeconfigString)
+	if err != nil {
+		return fmt.Errorf("failed to decode kubeconfig: %v", err)
+	}
+
+	// Step 4: Write kubeconfig to disk
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return fmt.Errorf("failed to get home directory: %v", err)
+	}
+
+	// Create directory if it doesn't exist
+	byohDir := filepath.Join(homeDir, service.ByohConfigDir)
+	if err := os.MkdirAll(byohDir, service.DefaultDirPerms); err != nil {
+		return fmt.Errorf("failed to create directory: %v", err)
+	}
+
+	// Write decoded kubeconfig
+	kubeconfigPath := filepath.Join(byohDir, "config")
+	if err := os.WriteFile(kubeconfigPath, decodedKubeconfig, service.DefaultFilePerms); err != nil {
+		return fmt.Errorf("failed to write kubeconfig: %v", err)
+	}
+
+	utils.LogSuccess("Successfully saved kubeconfig to %s", kubeconfigPath)
+	return nil
+}
+
+// CheckDNSResolution verifies that DNS resolution works for the FQDN
+func (c *K8sClient) CheckDNSResolution() ([]string, error) {
+	utils.LogInfo("Verifying DNS resolution for %s", c.fqdn)
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	var r net.Resolver
+	addrs, err := r.LookupHost(ctx, c.fqdn)
+	if err != nil {
+		return nil, fmt.Errorf("DNS resolution failed for %s: %v", c.fqdn, err)
+	}
+
+	if len(addrs) == 0 {
+		return nil, fmt.Errorf("DNS resolution returned empty result for %s", c.fqdn)
+	}
+
+	utils.LogSuccess("DNS resolution successful: %v", addrs)
+	return addrs, nil
+}
+
+// RunByohAgent sets up and starts the BYOH agent
+func (c *K8sClient) RunByohAgent(secretName string) error {
```

Comment:
Why is this function in the k8s package? Shouldn't this be in main?

Reply:
Moved it to onboard.go

Reply:
ohh yes, I've changed it to the onboard.go file